src/proj_equirect.jl

```diff
@@ -0,0 +1,419 @@
+    nθ, nφ = f.Ny, f.Nx
+    @assert nφ½₊1 == nφ÷2+1
+    @assert 2nθ   == nθₓ2
+
+    pθk = similar(f.arr, nθ, nφ)
+    for ℓ = 1:nφ½₊1
+        if (ℓ==1) | ((ℓ==nφ½₊1) & iseven(nφ))
+            pθk[:,ℓ] .= f.arr[1:nθ,ℓ]
+        else
+            pθk[:,ℓ]  .= f.arr[1:nθ,ℓ]
+            pθk[:,Jperm(ℓ,nφ)] .= conj.(f.arr[nθ+1:2nθ,ℓ])
+        end
+    end
+    EquiRectQUMap(m_ifft(pθk, 2) .* √nφ, f.metadata)
+end
+
+Base.getindex(f::EquiRectS0, ::typeof(!)) = AzFourier(f).arr
+Base.getindex(f::EquiRectS2, ::typeof(!)) = QUAzFourier(f).arr
+
+Base.getindex(f::EquiRectS0, ::Colon) = Map(f).arr
+Base.getindex(f::EquiRectS2, ::Colon) = QUMap(f).arr
+
+function Base.summary(io::IO, f::EquiRectField)
+    @unpack Ny,Nx,Nbatch = f
+    print(io, "$(length(f))-element $Ny×$Nx$(Nbatch==1 ? "" : "(×$Nbatch)")-pixel ")
+    Base.showarg(io, f, true)
+end
+
+function Base.summary(io::IO, f::EquiRectAzFourier)
+    @unpack Ny,Nx,Nbatch = f
+    print(io, "$(length(f))-element $Ny×$(Nx÷2+1)$(Nbatch==1 ? "" : "(×$Nbatch)")-pixel ")
+    Base.showarg(io, f, true)
+end
+
+function Base.summary(io::IO, f::EquiRectQUAzFourier)
+    @unpack Ny,Nx,Nbatch = f
+    print(io, "$(length(f))-element $(2Ny)×$(Nx÷2+1)$(Nbatch==1 ? "" : "(×$Nbatch)")-pixel ")
+    Base.showarg(io, f, true)
+end
+
+# block-diagonal operator
+# ================================================
+
+# ## Constructors
+
+function BlockDiagEquiRect{B}(block_matrix::A, proj::P) where {B<:Basis, P<:ProjEquiRect, T, A<:AbstractArray{T}}
+    BlockDiagEquiRect{B,P,T,A}(block_matrix, proj)
+end
+
+# Allows construction by a vector of blocks
+function BlockDiagEquiRect{B}(vector_of_blocks::Vector{A}, proj::P) where {B<:Basis, P<:ProjEquiRect, T, A<:AbstractMatrix{T}}
+    block_matrix = Array{T}(undef, size(vector_of_blocks[1])..., length(vector_of_blocks))
+    for b in eachindex(vector_of_blocks)
+        block_matrix[:,:,b] .= vector_of_blocks[b]
+    end
+    BlockDiagEquiRect{B}(block_matrix, proj)
+end
+
+# ## Linear Algebra basics 
+
+*(M::BlockDiagEquiRect{B}, f::EquiRectField) where {B<:Basis} = M * B(f)
+
+function *(M::BlockDiagEquiRect{AzFourier}, f::EquiRectAzFourier)
+    promote_metadata_strict(M.proj, f.proj) # ensure same projection
+    EquiRectAzFourier(@tullio(Bf[p,iₘ] := M.blocks[p,q,iₘ] * f.arr[q,iₘ]), f.metadata)
+end
+
+function *(M::BlockDiagEquiRect{QUAzFourier}, f::EquiRectQUAzFourier)
+    promote_metadata_strict(M.proj, f.proj) # ensure same projection
+    EquiRectQUAzFourier(@tullio(Bf[p,iₘ] := M.blocks[p,q,iₘ] * f.arr[q,iₘ]), f.metadata)
+end
+
+# TODO: Figure out how reduce duplication so I can define methods like this ...
+
+# function *(M::BlockDiagEquiRect{T}, f::EquiRect{T}) where {T<:Az}
+# ...
+
+# ## mapblocks for fun.(Mblocks,eachcol(f))
+
+function mapblocks(fun::Function, M::BlockDiagEquiRect{B}, f::EquiRectField) where {B<:Basis} 
+    mapblocks(fun, M, B(f))
+end
+
+function mapblocks(fun::Function, M::BlockDiagEquiRect{QUAzFourier}, f::EquiRectQUAzFourier)
```

Comment:
I noticed your earlier comment about wanting `mapslices` used here (so it will be optimized on a GPU). I can't find that comment tho ... I'm guessing you realized that `mapslices` doesn't work for this case (at least that I know of). However, I just wanted to remark that my intended usage of `mapblocks` is simply for convenient CPU preparation of the `BlockDiagEquiRect's` that will eventually be moved over to the GPU.  So I'm not really optimizing for speed here. Just simplicity and ease of use.

Reply:
> I'm guessing you realized that mapslices doesn't work for this case

Yea exactly (hence I deleted that comment)

> mapblocks is simply for convenient CPU preparation of the BlockDiagEquiRect's that will eventually be moved over to the GPU

Yep, definitely the right approach.